File: ProblemSets/Week1/mario.c
#include <cs50.h>
#include <stdio.h>

int main(void)
{
    // Prompt the user for height
    int height;
    do
    {
        height = get_int("Height: ");
    }
    while (!(height >= 1 && height <= 8));

    int startPadding = height - 1;
    for (int i = 0; i < height; i++)
    {
        for (int k = 0; k < startPadding; k++)
        {
            printf(" ");
        }

        for (int j = 0; j < 2 * (i + 2); j++)
        {
            if (j > i && j < i + 3)
            {
                printf(" ");
            }
            else
            {
                printf("#");
            }
        }
        startPadding--;
        printf("\n");
    }
}

File: ProblemSets/Week6/mario.py
from cs50 import get_int

height = -1

# Prompt user for height
while not (height > 0 and height <= 8):
    height = get_int("Height: ")

padding = height - 1
for i in range(height):
    # Prints left side of pyramid
    print(' ' * (padding) + '#' * (i + 1), end="")
    # Print space between halves
    print('  ', end="")
    # Prints right side of pyramid
    print('#' * (i + 1))
    padding -= 1

# Idea of char arithmetics based on the explanation
# about char and * operator from
# https://java2blog.com/print-character-n-times-python/
